src/towncrier/test/test_build.py

```diff
@@ -740,12 +740,25 @@ def test_title_format_false(self):
                     [tool.towncrier]
                     package = "foo"
                     title_format = false
+                    template = "template.rst"
                 """))
             os.mkdir("foo")
             os.mkdir("foo/newsfragments")
-            # Towncrier ignores .rst extension
-            with open("foo/newsfragments/124.feature.rst", "w") as f:
-                f.write("Extends levitation")
+            with open("template.rst", "w") as f:
+                f.write(dedent("""\
+                    Here's a hardcoded title added by the template
+                    ==============================================
+                    {% for section in sections %}
+                    {% set underline = "-" %}
+                    {% for category, val in definitions.items() if category in sections[section] %}
```

Comment:
it looks like all these jinja directives from the template are not used in this test.

I don't understand why you would like to have `title_format = false` **and** a custom template.
With a custom template you can just don't include the title part without having to use False.

Reply:
Yes, this template should be able to have all these normal template bits removed.  Change incoming.

Does it matter why you would want that scenario if it is documented and released functionality?  Or is this question more about making sure we have a common understanding of the feature than questioning if the situation should be tested?  I haven't really thought through what I think the configuration design ought to be as I'm targeting a bugfix+feature release without introducing breaking changes (nor deprecations).

None the less, false is exactly for the case where you specify a template that provides its own title.  If you don't specify false then you will (should?  i think?...  `:[`) get the default title before whatever your template does.  I do feel funny about the existing non-empty string or empty string/unspecified or false possibilities for `title_format`.  I guess I see the use cases along the lines of 'default title format and template' (unspecified or empty string), 'the default template is fine but i want to change just the title format' (non-empty string), and 'gosh darn it let me just write everything in my custom template' (false).

Hopefully this helps...  this is definitely an area that strikes me as simply confusing.

Reply:
> Does it matter why you would want that scenario if it is documented and released functionality? Or is this question more about making sure we have a common understanding of the feature than questioning if the situation should be tested? 

It's more about understanding the feature and its valid use cases.

I imagine someone reading this test 1 year in the future, looking at a bug and will say: Why you need this feature with a custom template.

So I prefer to have a test in which the feature is tested like in real file, and not using a combination of configuration option that  are possible, but will never be used.

At the same time, I think that this test is wrong. We should have a full test for the default template itself.

This should also be a test for the `{% if render_title %}` part of the default template